src/Microsoft.AspNet.Http/ParsingHelpers.cs

```diff
@@ -592,8 +588,10 @@ private static string DeQuote(string value)
 
             const NumberStyles styles = NumberStyles.AllowLeadingWhite | NumberStyles.AllowTrailingWhite;
             long value;
-            var rawValue = headers[HeaderNames.ContentLength];
```

Comment:
This should not throw. That's one of the reasons we introduced IReadableStringCollection in the first place. Indexers throwing for missing entries is annoying as hell.